Update readme for Java 17

vpa1977 opened the pull request:
#### Changes proposed in this pull request:
 - Update readme for Java 17


- [*] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

*Picture of a cool ROCK:*

vpa1977 committed:
update readme for Java 17

vpa1977 committed:
Update README.md - spelling

Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

cjdcordeiro commented on README.md:
```diff
@@ -3,3 +3,254 @@
+
+Eclipse Temurin images use `ubuntu:22.04` or `alpine:3.18` base images and extract Adoptium's build of Java 17 Runtime built with OpenJDK `legacy-jre-image` target. It contains hotspot binaries and a base list of modules required for the runtime and is generated from the JDK image with the `jlink` tool. It includes a class data-sharing cache.
+
+Azul Zulu packages a limited set of binaries and a full set of JDK modules. It includes class data-sharing cache. It only installs base dependencies (`libc` and `libnss`) and none of the Abstract Window Toolkit (AWT) such as fontconfig despite having AWT binaries present in the Java runtime.
```
```suggestion
Azul Zulu packages a limited set of binaries and a full set of JDK modules. It includes a class data-sharing cache. It only installs base dependencies (`libc` and `libnss`) and none of the Abstract Window Toolkit (AWT) such as `fontconfig` despite having AWT binaries present in the Java runtime.
```

cjdcordeiro commented on README.md:
```diff
@@ -3,3 +3,254 @@
+
+Amazon Corretto installs Java JDK package in `amazonlinux:2023` with all package dependencies. It packages a limited set of binaries and a full set of JDK modules.
+
+Google distroless image is built by installing Debian openjdk-17-jre-headless package and its dependencies and copying them into a scratch container. It contains a limited set of binaries and a full set of JDK modules.
```
```suggestion
The Google distroless image is built by installing the Debian openjdk-17-jre-headless package and its dependencies and copying them into a scratch container. It contains a limited set of binaries and a full set of JDK modules.
```

cjdcordeiro commented on README.md:
```diff
@@ -3,3 +3,254 @@
+
+Azul Zulu packages a limited set of binaries and a full set of JDK modules. It includes class data-sharing cache. It only installs base dependencies (`libc` and `libnss`) and none of the Abstract Window Toolkit (AWT) such as fontconfig despite having AWT binaries present in the Java runtime.
+
+Amazon Corretto installs Java JDK package in `amazonlinux:2023` with all package dependencies. It packages a limited set of binaries and a full set of JDK modules.
```
```suggestion
Amazon Corretto installs the Java JDK package in `amazonlinux:2023` with all package dependencies. It packages a limited set of binaries and a full set of JDK modules.
```

cjdcordeiro reviewed (changes requested):
Thanks for this ;)

I've left a few nitpicks.

Some of the text is also still pointing to the old `chiselled-jre` folder and image references, so that should be updated. 

I'm also adding @valentincanonical to the list of reviewers as I believe this is useful to him.

vpa1977 committed:
Update README.md

Co-authored-by: Cristovao Cordeiro <cristovao.cordeiro@canonical.com>
Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
Update README.md

Co-authored-by: Cristovao Cordeiro <cristovao.cordeiro@canonical.com>
Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
Update README.md

Co-authored-by: Cristovao Cordeiro <cristovao.cordeiro@canonical.com>
Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
Review comments

- capitalized Temurin
- ubuntu/chiselled_jre -> ubuntu/jre
- fixed article usage


Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
rename JRE 8 -> JRE17 in conclusion

Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

vpa1977 committed:
Implement grammarly style suggestions

Signed-off-by: Vladimir Petko <vladimir.petko@canonical.com>

cjdcordeiro reviewed (approved):
LGTM thanks. Let's wait for @valentincanonical

cjdcordeiro commented:
reminder (@valentincanonical)

cjdcordeiro commented:
@valentincanonical I'm merging this since it has been in review for a long time. please open an issue if you find anything that needs changing.
